### PR TITLE
[examples][cifar10] Add VGG like model, some refactoring

### PR DESCRIPTION
**Pull request by MatzeB**
*Description*:
Add a 2nd model to the cifar10 example that is close to VGG as implemented in pytorch vision vgg.py / vgg11. This is currently missing the `Dropout` layers as I couldn't find a builtin node for that in glow.

*Testing*:
Ran the cifar10 example with `-model-vgg`, accuracy is improving more steadily than with the jumpy behavior of the default model.

*Documentation*:
Related to #2111

### CODE REVIEW

**Comment by rdzhabarov** [comment]
> This is currently missing the Dropout layers as I couldn't find a builtin node for that in glow.

this is mostly node for training, as we are not training focused that's missing for now.

File: examples/cifar10.cpp [hunk 1]
```diff
@@ -46,6 +58,59 @@ llvm::cl::opt<BackendKind> executionBackend(
...
+  auto *RL2 = F->createRELU("relu", CV2);
+  auto *MP2 = F->createMaxPool("pool", RL2, 2, 2, 0);
+
+  auto *FCL1 = F->createFullyConnected(ctx, "fc", MP2, numLabels);
+  auto *SM = F->createSoftMax("softmax", FCL1, expected);
+  auto *save = F->createSave("ret", SM);
+  return save->getPlaceholder();
+}
+
+/// Creates a VGG Model (inspired by pytorch/torchvision vgg.py/vgg11
```
[the remark this answers is not in the document]
Comment (MatzeB) [reply examples/cifar10.cpp]: Added one.

File: examples/cifar10.cpp [hunk 2]
```diff
@@ -46,6 +58,59 @@ llvm::cl::opt<BackendKind> executionBackend(
 /// Size: (1 + (32 * 32 * 3)) * 10000 = 30730000.
 const size_t cifarImageSize = 1 + (32 * 32 * 3);
 const size_t cifarNumImages = 10000;
+const unsigned numLabels = 10;
+
+static Placeholder *createDefaultModel(Context &ctx, Function *F,
+                                       NodeValue input, NodeValue expected) {
+  // Create the rest of the network.
```
[the remark this answers is not in the document]
Comment (MatzeB) [reply examples/cifar10.cpp]: Artifact of moving code around, dropped it.

**Comment by nadavrot** [comment]
@MatzeB @rdzhabarov 

Adding VGG is a nice feature. We can also fix the 'simple' network. This [1] demo from ConvNetJS uses a simple neural network to train and infer Cifar. I think that a similar neural network should work for this demo app. It should be easy to copy the network structure from the short description in the link. This would fix the 'simple' network. 

[1] https://cs.stanford.edu/people/karpathy/convnetjs/demo/cifar10.html

**Comment by nadavrot** [comment]
@MatzeB @rdzhabarov Do you think that we can simplify the code here?
